stix2/utils.py

```diff
@@ -12,15 +12,34 @@
 NOW = object()
 
 
+class STIXdatetime(dt.datetime):
+    def __new__(cls, *args, **kwargs):
+        precision = kwargs.pop('precision', None)
+        if isinstance(args[0], dt.datetime):  # Allow passing in a datetime object
+            dttm = args[0]
+            args = (dttm.year,)
+            kwargs['month'] = dttm.month
+            kwargs['day'] = dttm.day
+            kwargs['hour'] = dttm.hour
+            kwargs['minute'] = dttm.minute
+            kwargs['second'] = dttm.second
+            kwargs['microsecond'] = dttm.microsecond
+            kwargs['tzinfo'] = dttm.tzinfo
+        self = dt.datetime.__new__(cls, *args, **kwargs)
+        self.precision = precision
+        return self
```

Comment:
I wish we didn't need to do this attribute-by-attribute copying. I knew there was a `timetuple()` method on `datetime` instances, but I just realized that it doesn't do exactly what we need, particularly with `tzinfo`. 

Is there any reason we couldn't just use (something like) the following?
```
     precision = kwargs.pop('precision', None)
     if isinstance(args[0], dt.datetime):  # Allow passing in a datetime object
         dttm = args[0]
     else:
         dttm = dt.datetime.__new__(cls, *args, **kwargs)
     dttm.precision = precision
     return dttm
```